src/test/java/com/github/dockerjava/core/command/FrameReaderITest.java

```diff
@@ -39,18 +47,17 @@ public void deleteDockerContainerImage() throws Exception {
 
     @Test
     public void canCloseFrameReaderAndReadExpectedLines() throws Exception {
-
         // wait for the container to be successfully executed
         int exitCode = dockerClient.waitContainerCmd(dockerfileFixture.getContainerId())
                 .exec(new WaitContainerResultCallback()).awaitStatusCode();
         assertEquals(0, exitCode);
 
-        Iterator<Frame> response = getLoggingFrames().iterator();
-
-        assertEquals(response.next(), new Frame(StreamType.STDOUT, "to stdout\n".getBytes()));
-        assertEquals(response.next(), new Frame(StreamType.STDERR, "to stderr\n".getBytes()));
-        assertFalse(response.hasNext());
-
+        final List<Frame> loggingFrames = getLoggingFrames();
+        final Frame outFrame = new Frame(StreamType.STDOUT, "to stdout\n".getBytes());
+        final Frame errFrame = new Frame(StreamType.STDERR, "to stderr\n".getBytes());
+        
+        assertThat(loggingFrames, containsInAnyOrder(outFrame, errFrame));
```

Comment:
when bash sends messages without delays it may happen that err will come before stdout. In my env it was always in different order so i switched to such test.